ludwig/schema/utils.py

```diff
@@ -51,10 +51,15 @@ def load_trainer_with_kwargs(
     In particular, it chooses the correct default type for an incoming config (if it doesn't have one already), but
     otherwise passes all other parameters through without change.
     """
-    from ludwig.constants import MODEL_ECD
-    from ludwig.schema.trainer import ECDTrainerConfig, GBMTrainerConfig
-
-    trainer_schema = ECDTrainerConfig if model_type == MODEL_ECD else GBMTrainerConfig
+    from ludwig.constants import MODEL_ECD, MODEL_GBM, MODEL_LLM
+    from ludwig.schema.trainer import ECDTrainerConfig, GBMTrainerConfig, LLMTrainerConfig
+
+    if model_type == MODEL_ECD:
```

Comment:
Good candidate for a registry pattern. Maybe as a TODO.

Reply:
Done